### DISCUSSION TITLE
Possible bug in bin sizes

### DISCUSSION THREAD

**Discussion by gevro** [etal/cnvkit#997] [General]
Hi,
I ran the latest version of cnvkit via docker with these settings:

cnvkit.py batch -n -f REF.fasta -m hybrid -g ACCESSIBLEREGIONS.bed -t PROBESBED.bed --target-avg-size 50000 --antitarget-avg-size 4000000 --antitarget-min-size 1000000

However, it is producing a target.bed file whose average bin size is only 122.79 bp. This should not be possible based on the above settings.

This is with hg38.

So there appears to be a bug in the way batch interprets these parameters:
--target-avg-size 50000

Or is this because bins cannot encompass more than one target? If so, I assume then that the segment step integrates information across multiple bins?

Thanks

**Comment by etal** [comment]
You got it, the target bin boundaries are limited by the regions in your `PROBESBED.bed`. If your panel captures exons with an average size of 122.79 bp, that will be the average in the output. The `--target-avg-size` parameter works the way you would think in WGS and in large captured regions like introns (e.g. if your panel is looking for fusion sites, not just the exome/exons).

Segmentation is where the signal is extracted across multiple adjacent targets and antitargets, so segments can be up to the length of a whole chromosome.
